src/main/java/com/github/lombrozo/jsmith/antlr/rules/RuleBlock.java

```diff
@@ -33,14 +37,53 @@
  * }
  * @since 0.1
  */
-public final class RuleBlock extends Unimplemented {
+@SuppressWarnings("PMD.OnlyOneConstructorShouldDoInitialization")
+public final class RuleBlock implements Rule {
+    /**
+     * Parent rule.
+     */
+    private final Rule top;
+
+    /**
+     * RuleAltList.
+     */
+    private final Rule list;
```

Comment:
@RechnoyBobr I't better to use `AtomicReference<Rule> list` here. Then, in `append` method you will be able to set it. In the construction you will be able to use `new AtomicReference()`.